network/simulations/discovery/discovery_test.go

```diff
@@ -147,7 +147,7 @@ func TestDiscoverySimulationSimAdapter(t *testing.T) {
 	testDiscoverySimulationSimAdapter(t, *nodeCount, *initCount)
 }
 
-func TestDiscoveryPersistenceSimulationSimAdapter(t *testing.T) {
+func XTestDiscoveryPersistenceSimulationSimAdapter(t *testing.T) {
```

Comment:
@zelig could you maybe add a few comments on what this test is supposed to test?

I am also not sure how the vars `persistenceEnabled` and `discoveryEnabled` are supposed to work - what if we tests are run concurrently?

It seems to me that this test is starting a network and connecting nodes in a chain, and then waiting for the Kademlia table to be setup by `hive`. At this point it stops the nodes, stops discovery and starts them back up, expecting for the Kademlia table to be setup `without discovery`.

If this is correct, it seems like we are changing the meaning of the `NoDiscovery` flag - previously it was just preventing `subPeerMsg` message exchanges, and not preventing actual connections, whereas now if we set the `NoDiscovery` flag to true, we don't want `hive` to trigger any connections what-so-ever.

Therefore if my assumption on this test is correct, it is expected for it to fail after the change.

What do you think?

Reply:
well you seemed to want to redefine this flag to not use the address book that is saved just allow manual (or snapshot driven) connection. 

This test is supposed to test that peers persist across sessions, i.e., that the addressbook is saved and used when bootstrapping connectivity

Reply:
BTW the addressbook persistence is broken for a long time - if there are peers that no longer exist, we never clear them out, and we keep on trying to connect to them indefinitely - something we tried fixing with @homotopycolimit at some point, but didn't work nicely as a `quick hack`.

Reply:
@nonsense this test was my first coding task on swarm so i take responsibility :D

Discovery should indeed not impact on whether we are connecting to peer from our address book, so I don't see why this should impact the way we connect to existing known peers, and I'm not sure that disabling the test would help this fact at all

Reply:
In fact, it makes more sense to have `--no-hive` to mean no connect loop. So you could simply fix this test by wrapping hive service in a struct and redefine its Protocols function to return empty. That way you can get the hive service with connect loop but no protocol running, so you can safely test bootstrapping from addressbook.

Reply:
I'd rather not redefine the `Protocols` function to test persistence of the kademlia address book, this seems like a very ugly hack.

With this PR `--no-hive-discovery` does not run the `connect()` loop.

Let's talk about the address book as part of the Kademlia epic, this is out of scope for this PR. Both me and Elad seem to think that similar to how bootnodes are connected to outside of `hive`, it makes sense to do something similar for the address book.